test/test-helpers.js

```diff
@@ -310,15 +310,16 @@ export const openMediaSource = function(player, clock) {
   // mock the tech *after* it has finished loading so that we don't
   // mock a tech that will be unloaded on the next tick
   mockTech(player.tech_);
-  player.tech_.hls.xhr = xhrFactory();
-
-  // simulate the sourceopen event
-  player.tech_.hls.mediaSource.readyState = 'open';
-  player.tech_.hls.mediaSource.dispatchEvent({
-    type: 'sourceopen',
-    swfId: player.tech_.el().id
-  });
-  clock.tick(1);
+  if (player.tech_.hls) {
```

Comment:
This helper assumed that hls would always be on tech, which this pr gets rid of.

Reply:
Won't it only get rid of it on dispose though?

Reply:
Yeah it should only get rid of it on dispose, but that will happen between sources as well.

Reply:
Let's prefer to return early